builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,378 @@
+}
+
+func (s *StepImportImage) createImageImportRequest(ctx context.Context, logger *PackerLogger) error {
+	logger.Info("Creating ContentLibraryItemImportRequest object %s in namespace %s.", s.ImportImageConfig.ImportRequestName, s.Namespace)
+
+	imageImportReq := &imgregv1.ContentLibraryItemImportRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ImportImageConfig.ImportRequestName,
+			Namespace: s.Namespace,
+		},
+		Spec: imgregv1.ContentLibraryItemImportRequestSpec{
+			Source: imgregv1.ContentLibraryItemImportRequestSource{
+				URL:            s.ImportImageConfig.ImportSourceURL,
+				SSLCertificate: s.ImportImageConfig.ImportSourceSSLCertificate,
+			},
+			Target: imgregv1.ContentLibraryItemImportRequestTarget{
+				Library: imgregv1.LocalObjectRef{
+					Kind:       ImportTargetKind,
+					APIVersion: ImportTargetAPIVersion,
+					Name:       s.ImportImageConfig.ImportTargetLocationName,
+				},
+				Item: imgregv1.ContentLibraryItemImportRequestTargetItem{
+					Name: s.ImportImageConfig.ImportTargetImageName,
+					Type: s.TargetItemType,
+				},
+			},
+		},
+	}
+
+	if err := s.KubeWatchClient.Create(ctx, imageImportReq); err != nil {
+		logger.Error("error creating the ContentLibraryItemImportRequest object %s.", imageImportReq.Name)
+		return err
+	}
+
+	logger.Info("Successfully created the ContentLibraryItemImportRequest object %s.", imageImportReq.Name)
+	return nil
+}
+
+func (s *StepImportImage) watchItemImport(ctx context.Context, state multistep.StateBag, logger *PackerLogger) error {
+	itemImportReqWatch, err := s.KubeWatchClient.Watch(ctx, &imgregv1.ContentLibraryItemImportRequestList{}, &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", s.ImportImageConfig.ImportRequestName),
+		Namespace:     s.Namespace,
+	})
+
+	if err != nil {
+		logger.Error("error watching the ContentLibraryItemImportRequest object in supervisor cluster")
+		return err
+	}
+
+	timedCtx, cancel := context.WithTimeout(ctx, time.Duration(s.ImportImageConfig.WatchImportTimeoutSec)*time.Second)
+
+	defer func() {
+		itemImportReqWatch.Stop()
+		cancel()
+
+		Mu.Lock()
+		IsWatchingImageImport = false
+		Mu.Unlock()
+	}()
+
+	Mu.Lock()
+	IsWatchingImageImport = true
+	Mu.Unlock()
+
+	for {
+		select {
+		case event := <-itemImportReqWatch.ResultChan():
+			if event.Object == nil {
+				return fmt.Errorf("watch ContentLibraryItemImportRequest event object is nil")
+			}
+
+			itemImportReqObj, ok := event.Object.(*imgregv1.ContentLibraryItemImportRequest)
+			if !ok {
+				return fmt.Errorf("failed to convert the watch ContentLibraryItemImportRequest event object")
+			}
+
+			if itemImportReqObj.Status.ItemRef != nil {
+				s.ImportItemResourceName = itemImportReqObj.Status.ItemRef.Name
+			}
+
+			importSuccess := false
+			for _, cond := range itemImportReqObj.Status.Conditions {
+				if cond.Type == imgregv1.ContentLibraryItemImportRequestComplete {
+					importSuccess = cond.Status == corev1.ConditionTrue
```

Comment:
Add a `break` here to avoid checking other conditions once we find import is successful?

Reply:
Sounds good.